### ISSUE TITLE
Burden testing: option --write-mask-snplist fails to return Indels which were part of the input annotation

### ISSUE THREAD

**Issue by ggstatgen** [rgcgithub/regenie#329]
Hi

I am running Regenie 3.1.1 for gene-level burden testing of quantitative phenotypes. 

In a previous run, I have found a gene-phenotype association of interest, so I've decided to run Regenie once more, this time using option

`--write-mask-snplist`

because I want to know what is the exact subset of variants I had provided in the input .annotations file which were tested in my Mask of interest at my MAF threshold of interest.

Now my mask of interest contained both SNPs and indels. The latter are 99% predicted LOFs frameshift. A typical annotation file I'm using looks like this

```
...
1:5712031:D:4 G_NAME    Lof
1:5713464:A:G G_NAME    missense
1:5713476:C:A G_NAME  missense
1:5713479:A:G G_NAME   missense
1:5713483:A:G G_NAME   missense_sp
1:5713526:D:1 G_NAME    Lof
...
```

And Regenie seems to run OK with this input - no error messages.

So I've then looked at the per-chr  .snplist file returned by regenie, and this, as I hoped contains one gene name.mask.MAF entry per line, followed by the list of variants tested in that mask.

However, I cannot find any indels in any of the tested masks in the .snplist file. There are no entries of the form

`1:5712031:D:4`

or

`1:5712031:I:4`

So for example if my Mask1 is 'LOF' the only lofs that I'm getting in the .snplist file output are SNP LOfs (stop inducing and splicing snps possibly). No 'indel' LOFs.

This too me can mean one of two things

1. Regenie is skipping testing my LOF indels altogether
2. Regenis is correctly including my LOF indels in the burden masks and testing them, however the produced .snplist file is not showing them.

Any help would be greatly appreciated.

**Comment by joellembatchou** [comment]
Hi,

Could you try with the most recent release? Some thoughts:
* Have you checked that the variants are not monomorphic in the input dataset (filtering down to the samples used in the analysis (ie if you run single variant test specifying `--minMAC 0.5`, do these variants show up in sumstats file)?
* Have you checked that the variant ID matches between annotation file and input genetic file for these variant (you can use `--check-burden-files` to see if there are ID mismatches)

Cheers,
Joelle

**Comment by ggstatgen** [comment]
Hi Joelle

Many thanks for your reply. It appears this was caused by a bug in my own code which generated the Regenie .annotations files for the burden tests. There was a mismatch between the way indels were represented in my genotype .bim files and the way I represented indels in my .annotations file. 

The above issue was invisible to me as I was processing my data on a cloud based platform and was calling regenie via a WDL pipeline. While I had the --check-burden-files option set, I did not explicitly instruct the WDL workflow to return the corresponding outputs, so all was smooth as far as I could tell. 

Many thanks for helping with this.
